python/foglamp/services/core/api/service.py

```diff
@@ -398,11 +410,82 @@ def load_c_plugin(plugin: str, service_type: str) -> Dict:
         :Example:
             curl -X GET http://localhost:8081/foglamp/service/installed
     """
-    services = []
-    svc_prefix = 'foglamp.services.'
-    for root, dirs, files in os.walk(_FOGLAMP_ROOT + "/" + "services"):
-        for f in files:
-            if f.startswith(svc_prefix):
-                services.append(f.split(svc_prefix)[-1])
-
+    services = get_service_installed()
     return web.json_response({"services": services})
+
+
+async def update_service(request: web.Request) -> web.Response:
+    """ update service
+
+    :Example:
+        curl -sX PUT http://localhost:8081/foglamp/service/Notification/notification/update
+    """
+    _type = request.match_info.get('type', None)
+    name = request.match_info.get('name', None)
+    try:
+        _type = _type.lower()
+        if _type != 'notification':
+            raise ValueError("Invalid plugin type. Must be 'notification'")
+
+        # Check requested service name is installed or not
+        installed_services = get_service_installed()
+        if name not in installed_services:
+            raise KeyError("{} service is not yet installed. So update is not possible.".format(name))
+
+        storage_client = connect.get_storage_async()
+        # TODO: process_name ends with "_c" suffix
+        payload = PayloadBuilder().SELECT("id", "enabled", "schedule_name").WHERE(['process_name', '=', '{}_c'.format(
+            _type)]).payload()
+        result = await storage_client.query_tbl_with_payload('schedules', payload)
+        sch_info = result['rows']
+        sch_list = []
+        if sch_info and sch_info[0]['enabled'] == 't':
+            status, reason = await server.Server.scheduler.disable_schedule(uuid.UUID(sch_info[0]['id']))
+            if status:
+                _logger.warning("{} service is disabled as {} service is updating..".format(
```

Comment:
I suggest `Disabled schedule for {}, as {} service of type {} is being updated...`
Don;t we need audit log?